documentation/API proposals/APIproposal_CapabilitiesandRuntimeRestrictions_T-MobileUS.md

```diff
@@ -0,0 +1,12 @@
+| **Field** | Description | 
+| ---- | ----- |
+| API family name | Capability and Runtime Restrictions |
+| API family owner | T-Mobile US |
+| API summary | CAMARA Service APIs are designed with many optional parameters and features.  It’s unreasonable to expect each API Producer (i.e. Operator) to support all these optional parameters.  In addition, some supported features and parameters may not be enabled at Service API invocation time, based on network state, who might be invoking and/or for whom/which device, location, …)  There is currently no mechanism to exchange such information with API Consumers (i.e. Application Service Providers (ASP)/Developers/Aggregators) and keep the CAMARA APIs the same across the Exposure Gateways. <br /> API Family is intended to cover the following areas:<br />1. Exchange of runtime restrictions (i.e. not supported parameters/features)<br />2. Exchange of capabilities (i.e. enabled/not enabled a set of parameters/features)<br />3. Topology exchange (i.e. abstraction) for capability-footprint association<br /><br />For the first area following examples can be given:<br />1. Device identifier in QoD can be of type Phone Number, IPv4 Address, IPv6 Address, Network Access Identifier (NAI).  One operator may support all of these identifiers in which case there will not be a need to list any restrictions towards the schema in the QoD API, however another operator supports only Phone Number, thus will need to inform ASPs not to use them.<br />2. If there is a regulation specific maximum accuracy level that must be set greater than the minimum Radius defined in ‘location-retrieval.yaml’, operator must be able to overwrite this new minimum.<br /><br />For the capabilities following example can be given:<br />1. While the operator may not support only one of the default set of QoD Profiles, at the time of invocation, one ore more of the supported QoD profiles may not be available due to network state and/or location.  Invocation of QoD with not-enabled QoD profile will result in error and lead to bad developer experience.  Operators must be able to efficiently exchange this information.|
```

Comment:
@jgarciahospital to reflect on the discussion on this API I'd like to update API Summary as follows:
    CAMARA Service APIs are designed with many optional properties and features. Some  features and properties may not be available at Service API invocation time, due to reasons including, but not limited, to network state, invoking entity, device/session in question, location etc. There is currently no mechanism to relay such information to API Consumers (i.e. Application Service Providers (ASP)/Developers/Aggregators) and keep the CAMARA APIs the same across the Exposure Gateways, avoid segmentation and recompilation of code due to changing conditions. <br/>
Proposed Family of APIs is a complementary Service Management API to all Service APIs and it is intended to cover the following areas:
    1. Exchange of runtime restrictions (i.e. not supported properties/features)
    2. Exchange of capabilities (i.e. enabled/not enabled a set of restrictions)
    3. Topology exchange (i.e. abstraction) for capability-footprint association 
    <br/>
For the first area following examples can be given: 
    1. Device identifier in QoD can be of type Phone Number, IPv4 Address, IPv6 Address, Network Access Identifier (NAI). One operator may support all of these identifiers in which case there will not be a need to list any restrictions towards the schema in the QoD API, however another operator supports only Phone Number, thus will need to inform ASPs not to use them.
    2. If there is a regulation specific maximum accuracy level that must be set greater than the minimum Radius defined in ‘location-retrieval.yaml’, operator must be able to overwrite this new minimum.
    <br/>
    
Note: Proposed Service Management APIs can be optionally implemented by API Providers, and in such instances can be optionally utilized by API Consumers.